sympy/physics/quantum/dagger.py

```diff
@@ -1,14 +1,15 @@
 """Hermitian conjugation."""
 
-from sympy.core import Expr, Mul
+from sympy.core import Expr
 from sympy.functions.elementary.complexes import adjoint
+from sympy.physics.quantum.operator import Operator
 
 __all__ = [
     'Dagger'
 ]
 
 
-class Dagger(adjoint):
+class Dagger(adjoint, Operator):
```

Comment:
Unfortunately, I don't think that it is a good idea to have multiple inheritance, and I think that it was also a bad idea that this class had inherited `adjoint` at all considering how printing of `adjoint` mutates when `Dagger` is imported.

Reply:
Oh man, there's something super nasty going on with `Dagger` vs `adjoint` here.

Reply:
Should the whole file just be replaced with `Dagger = adjoint`?

Reply:
> Oh man, there's something super nasty going on with `Dagger` vs `adjoint` here.

My thinking here was to inherit print functions from `adjoint` and other properties from `Operator`.

Reply:
> considering how printing of `adjoint` mutates when `Dagger` is imported

Opened gh-19849 about that